### PR TITLE
Np 2755 update publication context with url

### PR DESCRIPTION
**Pull request by svenngjeto**
Added linkedContext to Journal, Book, Report and Degree.

### CODE REVIEW

File: src/main/java/no/unit/nva/model/contexttypes/Book.java [hunk 1]
```diff
@@ -15,7 +17,7 @@
 import org.apache.commons.validator.routines.ISBNValidator;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
-public class Book implements BasicContext {
+public class Book implements BasicContext, LinkedContext {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/contexttypes/Book.java:20]: In an example of a BookAnthology and a Chapter, Would the Chapter have a linkedContext pointing to the BookAnthology ? If it is not this case but the opposite, we should have a collection of linkedContexts

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Comment by axthosarouris** [comment]
Requested change is only the docuementation.

**Comment by axthosarouris** [comment]
If linkedContext is a URI why is the setter a String?

File: src/main/java/no/unit/nva/model/contexttypes/Book.java [hunk 2]
```diff
@@ -164,10 +169,25 @@ public int hashCode() {
...
+    }
+
+    @Override
+    public void setLinkedContext(String linkedContext) {
+        setLinkedContext(URI.create(linkedContext));
+    }
+
+    protected void setLinkedContext(URI linkedContext) {
+        this.linkedContext = linkedContext;
+    }
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply src/main/java/no/unit/nva/model/contexttypes/Book.java]: Getters and setters reorganized together as they should have been...

File: src/main/java/no/unit/nva/model/contexttypes/Book.java (hunk 1 above)
Comment (svenngjeto) [reply src/main/java/no/unit/nva/model/contexttypes/Book.java:20]: Added an sample with a Book in a series, but there will/must be upcoming changes in the model regardin handling of series

File: src/main/java/no/unit/nva/model/contexttypes/LinkedContext.java [hunk 3]
```diff
@@ -4,7 +4,22 @@
 
 /**
  * LinkedContexts are contexts that are expressed as a URI, allowing an existing (external) Publication
- * to be used as a PublicationContext. For example, a BookAnthology containing a Chapter.
+ * to be used as a PublicationContext. For example, a BookAnthology containing a Chapter or a Book as part of a series.
+ *
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/contexttypes/LinkedContext.java:8]: This does not answer the basic question which is if the container has links to the contained or the contained a link to the container. A sentence of the form "e.g. A Chapter will have a linkedContext to the BookAnthology that contains it" or "A Book Anthology will have a linkedContext to the Chapter". I have understood that it must be the former but one cannot be too clear.

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: src/main/java/no/unit/nva/model/contexttypes/LinkedContext.java (hunk 3 above)
Comment (svenngjeto) [reply src/main/java/no/unit/nva/model/contexttypes/LinkedContext.java]: Tried to formulate some smart sentences to describe the funtionality

File: src/main/java/no/unit/nva/model/contexttypes/Book.java (hunk 1 above)
Comment (brinxmat) [reply src/main/java/no/unit/nva/model/contexttypes/Book.java:20]: The linkedContext is  always X isPartOf Y, not Y hasPart X

Comment (brinxmat) [reply src/main/java/no/unit/nva/model/contexttypes/Book.java:20]: Moreover, if we want to represent these things in terms of the thing that has the hasPart relationship, I consider this to be an index task as there is no guarantee that that we know the parts (example, the editor of an anthology, where all the contributors are not at Norwegian institutions)

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**